src/core/randomeval.js

```diff
@@ -0,0 +1,62 @@
+import seedrandom from 'seedrandom';
+
+function getrandomfn(ctx) {
+  let randomfn;
+  if (ctx.prngstate === undefined) {
+    // no call to a random function has been made.
+    // pre-populate the state info
+    randomfn = new seedrandom.alea(ctx.seed, { state: true });
+  } else {
+    randomfn = new seedrandom.alea('', { state: ctx.prngstate });
+  }
+  return randomfn;
+}
+
+export function randomctx(ctx) {
+  const r = getrandomfn(ctx);
+  const randomnumber = r();
+  const ctx2 = { ...ctx, prngstate: r.state() };
+  return { randomnumber, ctx: ctx2 };
+}
+
+export function addrandomop(G, fieldname, op) {
+  let rop = [{ op, fieldname }];
+  let _randomOps = [...(G._randomOps || []), ...rop];
+  return { ...G, _randomOps };
+}
+
+export function evaluaterandomops(G, ctx) {
+  let randomresults = {};
+  let ctx2 = ctx;
+
+  // some flow tests run without a defined G
+  if (G && G._randomOps !== undefined) {
+    G._randomOps.forEach(r => {
+      switch (r.op) {
+        case 'D6': {
```

Comment:
Use a constant named `RANDOM_D6` here?

Reply:
I've added a `SpotValue` object containing all common dice values from D4 to D20.